### ISSUE TITLE
Test Results as JSON

### ISSUE THREAD

**Issue by timmak** [lukeed/uvu#52]
Currently there doesn't seem to be a way to get the test result data other then parsing the cli output which is missing some information. Would be great if there either was a way to set a reporter or get an output that is parsable like tap.

**Comment by 43081j** [comment]
@lukeed whats your view on this? I want to write a TAP reporter for uvu too, i can either contribute it to uvu itself or write a third party util to convert uvu's output. of course i'd rather do the former, but you'd inevitably have to accept an increase in size for abstracting the reporter logic instead of the console logging you currently have. Happy to also contribute such an abstraction if you'll accept it.

**Comment by jakobrosenberg** [comment]
For scalability, I consider reporters/JSON a requirement. As with any other type of data, there's an inflection point on the data-size/complexity axis where tools become advantageous.

As projects grow, refactorings start to break the overview of the output. Often you just want to verify that a refactoring breaks the tests you expect it to break, but there's no summary or collapsable output, so you have to sift through walls of stack-traces, snapshot comparisons etc.

**Comment by 43081j** [comment]
its a bit of a lie if we don't implement reporters IMO.

if we implement JSON output, it just means the community will write tooling which consumes that output and produces the correct output. 

if we implement a basic reporter system (but only ship one reporter, the current one), it means the community will write reporters which produce the correct output. 

whether those inevitable tools exist in this repo or not doesn't really matter, but one of these things has to happen so the currently inadequate output can be made better.

**Comment by vsapronov** [comment]
Here's yet another view on this feature request. CircleCI is able to understand only junit tests reports (don't ask me why). If there would be some "techy" (like JSON) output available from uvu it would be very useful for building junit converter on top of it so that CircleCI can show tests results properly.

**Comment by vaelinn123** [comment]
Just leaving a note here, I was also looking to see if there were any reporters available with uvu to give me an alternate output. I was able to get useful json by passing the "json" reporter to c8 via: `c8 -r "json" uvu {path}`, c8 has a decent number of reporters. 

For me using the c8 reporters removes the need for a reporter specific to uvu as c8 is likely what will be ran in CI and vanilla uvu more frequently during local development where its current output would usually be more useful.

**Comment by 43081j** [comment]
weird, it doesn't output json though. so how did c8 manage to parse it?

**Comment by vaelinn123** [comment]
Yeah, it output json to the .coverage directory for me, albeit not very pretty json, but json nonetheless.

**Comment by 43081j** [comment]
ah i get you, yes it will be the coverage data.

there are two things here:

* test results
* coverage results

regardless of the output of the test results, im guessing c8 can still instrument the source and gather some coverage results.

but there's still no way to output useful test results (i.e. what passed/failed).
